zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/instance/UserTaskIntermediateState.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.instance;
+
+import io.camunda.zeebe.db.DbValue;
+import io.camunda.zeebe.msgpack.UnpackedObject;
+import io.camunda.zeebe.msgpack.property.ObjectProperty;
+import io.camunda.zeebe.protocol.impl.record.value.usertask.UserTaskRecord;
+
+/**
+ * This class holds the necessary user task data (including variables) to finalize commands such as
+ * `COMPLETE`, `UPDATE`, `ASSIGN` etc. mainly when task listeners are defined for these operations.
```

Comment:
🔧 This introduction could use an example for clarification. Something like:

> For example, when the user task is being assigned for the first time, the user task must remain unassigned while the task listener blocks the assignment. In the meantime, this tracks the new `assignee` which can be used when the task listener completes.

Reply:
Thanks for the suggestion, Nico!
I've added the example as per your feedback in this [commit](https://github.com/camunda/camunda/pull/22576/commits/65e06b31728360a30fcf5bc68a10bac29483f3fb#diff-495cacb2dfd86d4f86cddc70e7ce10b65ef46c2716d87dbbcc8e288f495a02d1R34-R37).